x-ui.sh

```diff
@@ -671,6 +671,102 @@ run_speedtest() {
     speedtest
 }
 
+iplimit_main() {
+    echo -e "${green}\t1.${plain} Install Fail2ban and configure IP Limit"
+    echo -e "${green}\t2.${plain} Uninstall"
+    echo -e "${green}\t3.${plain} Check logs"
+    read -p "Choose an option: " choice
+    case "$choice" in
+        1) 
+            confirm "Proceed?" "y"
+            if [[ $? == 0 ]]; then
+                install_iplimit
+            else
+                show_menu
+            fi ;;
+        2)  
+            read -p "Remove Fail2ban aswell? (Default:n) [y/n]: " temp
+            if [[ "${temp}" == "y" || "${temp}" == "Y" ]]; then
+                sudo systemctl disable fail2ban
+                sudo systemctl stop fail2ban
+                rm -f /etc/fail2ban/filter.d/3x-ipl.conf
+                rm -f /etc/fail2ban/action.d/3x-ipl.conf
+                sudo apt-get remove fail2ban -y
+            else
+                rm -f /etc/fail2ban/filter.d/3x-ipl.conf
+                rm -f /etc/fail2ban/action.d/3x-ipl.conf
+                sudo cp /etc/fail2ban/jail.conf /etc/fail2ban/jail.local
+                sudo systemctl restart fail2ban
+            fi
+            ;;
+        3)
+            cat /var/log/3xipl-banned.log ;;
+        *) echo "Invalid choice" ;;
+    esac
+}
+
+install_iplimit() {
+    if ! command -v fail2ban-client &>/dev/null; then
+        echo -e "${green}Fail2ban is not installed. Installing now...!${plain}\n"
+        sudo apt-get update
+        sudo apt-get install fail2ban -y
+    else
+        echo -e "${yellow}Fail2ban is already installed.${plain}\n"
+    fi
+
+    echo -e "${green}Configuring IP Limit...${plain}\n"
+    #Check if jail.local exists
+    if ! test -f "/etc/fail2ban/jail.local"; then
+        sudo cp /etc/fail2ban/jail.conf /etc/fail2ban/jail.local
+    fi
+
+    #Check if [3x-ipl] jail exists
+    if ! grep -qw '3x-ipl' /etc/fail2ban/jail.local; then
+        echo $'\n[3x-ipl]\nenabled=true\nfilter=3x-ipl\naction=3x-ipl\nlogpath=/var/log/daemon.log\nmaxretry=3\nfindtime=100\nbantime=300' >> /etc/fail2ban/jail.local
+    fi
+
+    #Check if 3x-ipl filter exist and remove if true
+    if test -f "/etc/fail2ban/filter.d/3x-ipl.conf"; then
+        rm -f /etc/fail2ban/filter.d/3x-ipl.conf
+    fi
+
+    #Check if 3x-ipl action exist and remove if true
+    if test -f "/etc/fail2ban/action.d/3x-ipl.conf"; then
+        rm -f /etc/fail2ban/action.d/3x-ipl.conf
+    fi
+
+    echo $'[Definition]\nfailregex = [LIMIT_IP].+Email= <F-USER>.+</F-USER>.+SRC= <HOST>\nignoreregex =' >> /etc/fail2ban/filter.d/3x-ipl.conf
```

Comment:
can you please modify this? Instead of using the operator (>>) to append the file, please use the operator (>) to overwrite the file and it will be safe. and, you can remove the lines that are meant for deleting the file.

```suggestion
    echo $'[Definition]\nfailregex = [LIMIT_IP].+Email= <F-USER>.+</F-USER>.+SRC= <HOST>\nignoreregex =' > /etc/fail2ban/filter.d/3x-ipl.conf
```